src/lib/services/client-metrics/last-seen/last-seen-service.ts

```diff
@@ -17,17 +17,26 @@ export class LastSeenService {
 
     private logger: Logger;
 
+    private lastSeenStore: ILastSeenStore;
+
     private featureToggleStore: IFeatureToggleStore;
 
+    private config: IUnleashConfig;
+
     constructor(
-        { featureToggleStore }: Pick<IUnleashStores, 'featureToggleStore'>,
+        {
+            featureToggleStore,
+            lastSeenStore,
+        }: Pick<IUnleashStores, 'featureToggleStore' | 'lastSeenStore'>,
         config: IUnleashConfig,
         lastSeenInterval = secondsToMilliseconds(30),
     ) {
+        this.lastSeenStore = lastSeenStore;
         this.featureToggleStore = featureToggleStore;
         this.logger = config.getLogger(
             '/services/client-metrics/last-seen-service.ts',
         );
+        this.config = config;
 
         this.timers.push(
             setInterval(() => this.store(), lastSeenInterval).unref(),
```

Comment:
shouldn't this be part of the scheduler?

Reply:
I will look into separating this in a follow-up